### PR TITLE
LF-60 templates in DB

### PR DESCRIPTION
**Pull request by jeffersondarcy**
replaces https://github.com/corona-school/backend/pull/452
Reading templates and variables:
```gql
query { 
	notifications {
    id
    description
    mailjetTemplateId
    active
    # All the message templates for different languages
  	messageTranslations { 
    	template
      language
    }
    # Variables that can be used in the templates
    variables
  }
}
```

Adding templates:
```gql
mutation { 
  notificationSetMessageTranslation(notificationId: 1 language: "de" 
    headline: "Wilkommen bei Lern-Fair {{user.firstname}}"
    body: "Dein Screening war erfolgreich, du bist nun Teil des Teams!"
    navigateTo: "/matching")
}
```
to pass the validation the sample context of notification 1 should contain user name
`{"user": {"firstname": MyName"}}`

### CODE REVIEW

**Review by Jonasdoubleyou — APPROVED** [review]
Very nice!

**Comment by Jonasdoubleyou** [comment]
You can also bring this change to master then. I'll take care of verifying and rolling it out next week.

File: common/notification/messages.ts [hunk 1]
```diff
@@ -0,0 +1,62 @@
...
+    const notification = await prisma.notification.findUnique({
+        where: { id: notificationId },
+        select: { type: true },
+    });
+    const translation = await prisma.message_translation.findFirst({
+        where: { notificationId: notificationId, language },
+        select: { template: true, navigateTo: true },
+    });
+
+    const { headline, body } = translation.template as any as TranslationTemplate; // @TODO: is it possible to fix any?
```
Comment (Jonasdoubleyou) [inline common/notification/messages.ts:22]: translation can be null no? So translation.template might fail?

File: common/notification/index.ts [hunk 2]
```diff
@@ -201,7 +208,24 @@ export async function checkReminders() {
...
-// TODO: function for user preferences "categories"
+// Renders a Message for a certain Concrete Notification
+export async function getMessage(
+    concreteNotification: ConcreteNotification | ConcreteNotificationPrisma,
+    language: TranslationLanguage = TranslationLanguage.DE
+): Promise<NotificationMessageType> {
+    const legacyUser = await getUserTypeORM(concreteNotification.userId);
+    const context = getContext(concreteNotification.context as NotificationContext, legacyUser);
+
+    const { type, headline, body, navigateTo } = await getMessageForNotification(concreteNotification.notificationID, language);
```
Comment (Jonasdoubleyou) [inline common/notification/index.ts:219]: Message can be null

File: graphql/concrete_notification/fields.ts [hunk 3]
```diff
@@ -40,10 +41,13 @@ export class ExtendedFieldsConcreteNotificationResolver {
         return await prisma.notification.findUnique({ where: { id: concreteNotification.notificationID } });
     }
 
-    @FieldResolver((returns) => NotificationMessage)
+    @FieldResolver((returns) => NotificationMessageType)
```
Comment (Jonasdoubleyou) [inline graphql/concrete_notification/fields.ts:44]: What happens when a Notification does not have a message maintained?

**Review by LomyW — APPROVED** [review]
nice work! 🎉

File: common/notification/messages.ts (hunk 1 above)
Comment (jeffersondarcy) [reply common/notification/messages.ts]: added some changes

File: common/notification/index.ts (hunk 2 above)
Comment (jeffersondarcy) [reply common/notification/index.ts]: fixed

File: graphql/concrete_notification/fields.ts (hunk 3 above)
Comment (jeffersondarcy) [reply graphql/concrete_notification/fields.ts]: fixed

**Review by Jonasdoubleyou — APPROVED** [review]

**Pull request merged**